cfme/utils/ssh.py

```diff
@@ -130,24 +130,21 @@ def __init__(self, stream_output=False, **connect_kwargs):
         self.f_stdout = connect_kwargs.pop('stdout', sys.stdout)
         self.f_stderr = connect_kwargs.pop('stderr', sys.stderr)
 
-        # load the defaults for ssh
-        default_connect_kwargs = {
-            'timeout': 10,
-            'allow_agent': False,
-            'look_for_keys': False,
-            'gss_auth': False
-        }
-        # Load credentials and destination from confs, if connect_kwargs is empty
-        if not connect_kwargs.get('hostname'):
```

Comment:
I think this was a bad pattern to follow - only setting default endpoint/credentials if hostname is not passed in.

Made the setting flatter in my update. Setup the entire dictionary of defaults, and then update that with anything that was passed in as connect_kwargs and has not been popped off in the attribute lookups at the start of the method.